### PR TITLE
feat(std/io): ReadableStream from AsyncIterator & WritableStream from Writer

### PR DESCRIPTION
**Pull request by crowlKats**
we should also rename `fromStreamWriter` & `fromStreamReader` to `writerFromStreamWriter` & `readerFromStreamReader`.
Going to add tests later

### CODE REVIEW

File: std/io/streams.ts [hunk 1]
```diff
@@ -34,3 +34,29 @@ export function fromStreamReader(
     },
   };
 }
+
+export function ReadableStreamFromAsyncIterator(
```
Comment (bartlomieju) [inline std/io/streams.ts:38]: Name should start with lowercase

File: std/io/streams.ts [hunk 2]
```diff
@@ -34,3 +34,29 @@ export function fromStreamReader(
...
+      if (done) {
+        controller.close();
+      } else {
+        controller.enqueue(value);
+      }
+    },
+  });
+}
+
+export function WritableStreamFromWriter(
```
Comment (bartlomieju) [inline std/io/streams.ts:54]: Ditto

File: std/io/streams.ts [hunk 3]
```diff
@@ -34,3 +34,29 @@ export function fromStreamReader(
...
+  writer: Deno.Writer,
+): WritableStream<Uint8Array> {
+  return new WritableStream({
+    async write(chunk) {
+      await writer.write(chunk);
```
Comment (bartlomieju) [inline std/io/streams.ts:59]: I'm pretty sure it's not guaranteed that `writer.write()` will write full chunk in one go - probably should be `Deno.writeAll(writer, chunk)`

**Review by bartlomieju — CHANGES_REQUESTED** [review]

**Comment by bartlomieju** [comment]
@crowlKats can you add some test cases?

File: std/io/streams.ts [hunk 4]
```diff
@@ -34,3 +34,31 @@ export function fromStreamReader(
...
+    async write(chunk) {
+      await Deno.writeAll(writer, chunk);
+    },
+  });
+}
+
+/** Create a `ReadableStream` from an `AsyncIterator`. */
+export function readableStreamFromAsyncIterator<T>(
+  iterator: AsyncIterableIterator<T>,
+): ReadableStream<T> {
```
Comment (nayeemrmn) [inline std/io/streams.ts:52]: IMO `readableStreamFromReader()` should exist either way. This module exists to provide bridges between Deno IO and whatwg streams. One should be able to "expect" a `readableStreamFromReader` looking at the other functions.

In this case, there's no problem with having both even if one would have a trivial implementation. Similar to `readLines()` in `std/io/bufio.ts`.

**Comment by ry** [comment]
I'm missing some context on this PR... what is the purpose of these renames?

**Comment by crowlKats** [comment]
This is not just renames, but rather 2 new functions: one allows to create a ReadableStream from an AsyncIterator and the other allows to create a WritableStream from a Deno.Writer.
The renames are there just because it would make differentiating the functions easier; `fromStreamWriter` is too vague, but `writerFromStreamWriter` gives enough context on what it does.

**Review by ry — APPROVED** [review]
LGTM - thanks @crowlKats

File: std/io/streams.ts (hunk 4 above)
Comment (crowlKats) [reply std/io/streams.ts:52]: @nayeemrmn i will look into it soon; the best usage for that would a bytes ReadableStream, but still not knowledgeable about those

**Pull request merged**